src/Hyphenator/HyphenationRule.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Hyphenator;
+
+class HyphenationRule
+{
+    private string $pattern;
+    private array $levels;
+
+    public function __construct(string $rule)
+    {
+        $validNumbers = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+        $this->levels = [];
+        $chars = str_split($rule);
+        if (strlen($rule) > 0 && $this->thisCharValid(0, $chars)) {
+            $this->levels[] = 0;
+        }
+        for($i = 0; $i < count($chars); $i++) {
```

Comment:
Nit pick, but I would but a vertical pace here, for better readability (visually easier on eyes).